python/demo/poisson/demo_poisson.py

```diff
@@ -175,14 +175,16 @@
 # represent the solution. (Upon initialization, it is simply set to the
 # zero function.) A :py:class:`Function
 # <dolfinx.functions.fem.Function>` represents a function living in
-# a finite element function space. Next, we can call the :py:func:`solve
-# <dolfinx.fem.solving.solve>` function with the arguments ``a == L``,
-# ``u`` and ``bc`` as follows: ::
+# a finite element function space. Next, we initialize a solver using the :py:class:`LinearVariationalSolver
+# <dolfinx.fem.linearvariationalsolver.LinearVariationalSolver>`.
+# This class is initialized with the arguments ``a == L``, ``u`` and ``bc`` as follows: ::
 
-# Compute solution
 u = Function(V)
-solve(a == L, u, bc, petsc_options={"ksp_type": "preonly", "pc_type": "lu"})
+lin_solver = LinearVariationalSolver(a == L, u, [bc])
 
+# When we want to compute the solution to the problem, we can specify what kind of solver we want to use.
+# In this problem, we use a direct LU solver, which is defined through the dictionary ``petsc_options``.
+lin_solver.solve(petsc_options={"ksp_type": "preonly", "pc_type": "lu"})
```

Comment:
These argument probably need to be passed to the `LinearVariationalSolver` object as the it should store its solver (ksp) object.